src/backend/metal/src/device.rs

```diff
@@ -1857,10 +1857,15 @@ impl hal::Device<Backend> for Device {
     }
 
     fn wait_idle(&self) -> Result<(), error::HostExecutionError> {
-        self.shared.queue_pool
-            .lock()
-            .unwrap()
-            .wait_idle(&self.shared.device);
+        debug!("waiting for idle");
```

Comment:
Any reason not to share the `wait_idle` implementation across device and command?

Reply:
it could be shared, yes